cmds/webboot/network.go

```diff
@@ -0,0 +1,151 @@
+package main
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	ui "github.com/gizak/termui/v3"
+	"github.com/u-root/NiChrome/pkg/wifi"
+	"github.com/u-root/webboot/pkg/menu"
+	"github.com/vishvananda/netlink"
+)
+
+func connected() bool {
+	client := http.Client{
+		Timeout: 10 * time.Second,
+	}
+
+	if _, err := client.Get("http://google.com"); err != nil {
+		return false
+	}
+	return true
+}
+
+func interfaceEntries() ([]menu.Entry, error) {
+	interfaces, err := netlink.LinkList()
+	if err != nil {
+		return nil, err
+	}
+
+	var ifEntries []menu.Entry
+	for _, iface := range interfaces {
+		ifEntries = append(ifEntries, &Interface{label: iface.Attrs().Name})
+	}
+	return ifEntries, nil
+}
+
+func interfaceIsWireless(ifname string) bool {
+	devPath := fmt.Sprintf("/sys/class/net/%s/wireless", ifname)
+	if _, err := os.Stat(devPath); err != nil {
+		return false
+	}
+	return true
+}
+
+func setupNetwork(uiEvents <-chan ui.Event) (bool, error) {
```

Comment:
I wondered would that be better if you do not call selectNetworkInterface inside the function, but send the name of interface as a parameter to this function? since the selectNetworkInterface is more focus on UI things, and the setupNetwork is actually more care about setting network. so after separating them this function could get rid of UI things. And maybe you can move this function to the new pkg.